inference_schema/parameter_types/standard_py_parameter_type.py

```diff
@@ -40,6 +40,21 @@ def __init__(self, sample_input):
                 if issubclass(type(data), AbstractParameterType):
                     self.sample_data_type_list.append(data)
 
+    def _get_supported_versions(self):
+        supported_list = ['3.0', '3.1']
+        v2_compatible = True
+        if self.sample_data_type is list or self.sample_data_type is tuple:
+            iterable_data_type = type(self.sample_input[0])
+            for data in self.sample_input:
+                if type(data) != iterable_data_type:
+                    v2_compatible = False
+        if v2_compatible:
+            supported_list += ['2.0']
+        return sorted(supported_list)
+
+    def supported_versions(self):
```

Comment:
Add this to the abstract type please, as a forcing function to make sure it ends up on all parameter types

Reply:
added - right now the non-standard-pyfunc types have essentially no-op implementations but that seems like a fine place to start

Reply:
We need implementations for pandas and numpy at least. The spark one I'm more okay with being a no-op, however since we need the others I'd probably say might as well do the spark one at the same time.

Reply:
Discussed - implementation is now centralized